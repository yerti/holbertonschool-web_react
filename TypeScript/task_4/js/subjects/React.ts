namespace Subjects {
  export class React extends Subject {
    experienceTeachingReact?: number;

    getRequirements(): string {
      return super.getRequirements() + "\nJavaScript library: React";
    }
  }
}
