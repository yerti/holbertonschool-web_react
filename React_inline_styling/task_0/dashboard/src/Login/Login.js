import React from 'react'

export default function Login() {
    return (
        <>
            <p>Login to access the full dashboard</p>
            <div>
                <label htmlFor="email">Email:</label>
                <input
                    type="email"
                    id="email"
                />

                <label htmlFor="password">Password:</label>
                <input
                    type="password"
                    id="password"
                />

                <button>OK</button>
            </div>
        </>
    )
}
