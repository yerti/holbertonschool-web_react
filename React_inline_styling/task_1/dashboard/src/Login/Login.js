import React from 'react'
import { StyleSheet, css } from 'aphrodite';

const styles = StyleSheet.create({
    contentLogin: {
        width: '300px',
        display: 'flex',
        flexDirection: 'column',
    },
    buttoLogin : {
        marginTop: '30px',
    }
})

export default function Login() {
    return (
        <>
            <p>Login to access the full dashboard</p>
            <div className={css(styles.contentLogin)}>
                <label htmlFor="email">Email:</label>
                <input
                    type="email"
                    id="email"
                />
                
                <label htmlFor="password">Password:</label>
                <input
                    type="password"
                    id="password"
                />

                <button className={css(styles.buttoLogin)}>OK</button>
            </div>
        </>
    )
}
