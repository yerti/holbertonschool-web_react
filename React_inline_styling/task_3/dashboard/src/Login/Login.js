import React from 'react'
import { StyleSheet, css } from 'aphrodite';

const styles = StyleSheet.create({
    contentLogin: {
        width: '300px',
        display: 'flex',
        flexDirection: 'column',
    },
    buttoLogin : {
        marginTop: '30px',
    },
    smallButtonLogin: {
        width: '40px',
        height: '25px',
        border: '2px solid #F9B572',
        borderRadius: '4px',
    }
})

export default function Login() {
    return (
        <>
            <p>Login to access the full dashboard</p>
            <div className={css(styles.contentLogin)}>
                <label htmlFor="email">Email:</label>
                <input
                    type="email"
                    id="email"
                />
                
                <label htmlFor="password">Password:</label>
                <input
                    type="password"
                    id="password"
                />

                <button className={css(styles.buttoLogin, styles.smallButtonLogin)}>OK</button>
            </div>
        </>
    )
}
