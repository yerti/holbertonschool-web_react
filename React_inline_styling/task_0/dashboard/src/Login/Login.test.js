import React from "react";
import { shallow } from "enzyme";
import Login from "./Login";


describe('Unitest Login Component', () => {
    it('Shallow render the Login component to verify it renders without crashing', () => {
        const wrapper = shallow(<Login />);
        expect(wrapper).toMatchSnapshot();
    });

    it('render 2 input tags and 2 label tags', () => {
        const wrapper = shallow(<Login />);
        expect(wrapper.find('input')).toHaveLength(2);
        expect(wrapper.find('label')).toHaveLength(2);
    });
});
