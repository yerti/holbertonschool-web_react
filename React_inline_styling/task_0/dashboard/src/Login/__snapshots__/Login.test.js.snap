// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unitest Login Component Shallow render the Login component to verify it renders without crashing 1`] = `ShallowWrapper {}`;
